Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -16,54 +17,81 @@ public sealed class PuddleSystem : EntitySystem
+        /// </summary>
+        /// <param name="uid">Puddle we have initialized</param>
+        /// <param name="puddle">PuddleComponent of the puddle we have initialized</param>
+        /// <param name="appearance">AppearanceComponent of the puddle we have initialized</param>
+        private void InitializeAppearance(EntityUid uid,
+            PuddleComponent? puddle = null,
+            AppearanceComponent? appearance = null)
         {
-            if (Deleted(puddleComponent.Owner) || EmptyHolder(uid, puddleComponent) ||
-                !EntityManager.TryGetComponent<AppearanceComponent>(uid, out var appearanceComponent))
+            if (!Resolve(uid, ref puddle, ref appearance, false))
+                return;
+
+            appearance.SetData(PuddleVisuals.VisualSeed, visualSeed); // This only needs to be set once, when the puddle is initialized.
+
+            UpdateAppearance(uid, puddle);
+        }
+
+        private void UpdateAppearance(EntityUid uid,
+            PuddleComponent? puddle = null,
+            AppearanceComponent? appearance = null)
+        {
+            if (!Resolve(uid, ref puddle, ref appearance, false)
+                || EmptyHolder(uid, puddle))
             {
                 return;
             }
 
             // Opacity based on level of fullness to overflow
             // Hard-cap lower bound for visibility reasons
-            var volumeScale = puddleComponent.CurrentVolume.Float() / puddleComponent.OverflowVolume.Float() * puddleComponent.OpacityModifier;
-            var puddleSolution = _solutionContainerSystem.EnsureSolution(uid, puddleComponent.SolutionName);
 
+            var volumeScale = puddle.CurrentVolume.Float() / puddle.OverflowVolume.Float() * puddle.OpacityModifier;
+            var puddleSolution = _solutionContainerSystem.EnsureSolution(uid, puddle.SolutionName);
 
-            bool hasEvaporationComponent = EntityManager.TryGetComponent<EvaporationComponent>(uid, out var evaporationComponent);
-            bool canEvaporate = (hasEvaporationComponent &&
-                                (evaporationComponent.LowerLimit == 0 || puddleComponent.CurrentVolume > evaporationComponent.LowerLimit));
+            bool useWetFloorEffect = false; // Assume it won't by default, then check if it will.
 
-            // "Does this puddle's sprite need changing to the wet floor effect sprite?"
-            bool changeToWetFloor = (puddleComponent.CurrentVolume <= puddleComponent.WetFloorEffectThreshold
-                                    && canEvaporate);
+            if (TryComp(uid, out EvaporationComponent? evaporation) && evaporation.EvaporationToggle// if puddle is evaporating.
+                && puddle.CurrentVolume <= puddle.WetFloorEffectThreshold) // if puddle's volume is below a certain threshold. TODO: By default this threshold is hardcoded at what a standard mop can pick up -- need to make this more robust.
+            {
+                useWetFloorEffect = true;
+            }
```

Comment:
If the point is to switch to wet floor sparkles to indicate to a janitors when their job is done, shouldn't this also check if 
`evaporation.LowerLimit == 0 && puddle.CurrentVolume < evaporation.UpperLimit`?

Also note that `puddle.CurrentVolume` isn't cached, its actually doing a EntitySystem.Get and a `_solutionContainerSystem.TryGetSolution` every time it gets accessed. So if you use it more than once, you should the result in a local variable.

Reply:
> If the point is to switch to wet floor sparkles to indicate to a janitors when their job is done, shouldn't this also check if `evaporation.LowerLimit == 0 && puddle.CurrentVolume < evaporation.UpperLimit`?

Ah, yes, that's essentially what `evaporation.EvaporationToggle` does. I introduced it in  #6584 to cover any case where evaporation would need to be paused for whatever reason. Right now it only checks for `LowerLimit` and `UpperLimit`, but this way we can add more nuance to `EvaporationToggle` without having to account for those changes in systems like `PuddleSystem`, which only care about whether evaporation is active.

>Also note that puddle.CurrentVolume isn't cached, its actually doing a EntitySystem.Get and a _solutionContainerSystem.TryGetSolution every time it gets accessed. So if you use it more than once, you should the result in a local variable.

Good point! Didn't realize that. I'll call it once and store the result in a local variable then.

Reply:
I think I'll put the `&& evaporation.EvaporationToggle` on its own line for clarity. Must have just missed it, usually I do one line per condition so it's plain to see.